skill-tree/src/test/java/com/RDS/skilltree/unit/EndorsementServiceTest.java

```diff
@@ -60,6 +80,431 @@ public void itShouldGetEndorsementsById() {
+                                pageRequest, "invalid-skill-id", UUID.randomUUID().toString()));
+    }
+
+    @Test
+    @DisplayName(
+            "Return paginated result having 2 pages when number of endorsements with a given userID is 15")
+    @Disabled
+    public void itShouldReturnPaginatedResultOnSearch() throws IOException {
+        PageRequest pageRequest = PageRequest.of(0, 10);
+        String userID = UUID.randomUUID().toString();
+
+        List<EndorsementModelFromJSON> dummyEndorsements = new ArrayList<>();
+        for (int i = 0; i < 15; i++) {
+            dummyEndorsements.add(
+                    new EndorsementModelFromJSON(
+                            UUID.randomUUID(),
+                            UUID.fromString(userID),
+                            UUID.randomUUID(),
+                            "APPROVED",
+                            LocalDateTime.now(),
+                            UUID.randomUUID(),
+                            LocalDateTime.now(),
+                            UUID.randomUUID()));
+        }
+
+        when(objectMapper.readValue(
+                        ArgumentMatchers.<InputStream>any(),
+                        ArgumentMatchers.<TypeReference<List<EndorsementModelFromJSON>>>any()))
+                .thenReturn(dummyEndorsements);
+        Page<EndorsementModelFromJSON> result =
+                endorsementService.getEndorsementsFromDummyData(pageRequest, null, userID);
+
+        assertEquals(2, result.getTotalPages());
+        assertEquals(15, result.getTotalElements());
+    }
+
+    @Test
+    @DisplayName("Return empty page when accessing out of bound page in paginated result")
+    @Disabled
+    public void itShouldReturnEmptyPaginatedResultOnSearch() throws IOException {
+        PageRequest pageRequest = PageRequest.of(10, 10);
+        String userID = UUID.randomUUID().toString();
+
+        List<EndorsementModelFromJSON> dummyEndorsements = new ArrayList<>();
+        for (int i = 0; i < 15; i++) {
+            dummyEndorsements.add(
+                    new EndorsementModelFromJSON(
+                            UUID.randomUUID(),
+                            UUID.fromString(userID),
+                            UUID.randomUUID(),
+                            "APPROVED",
+                            LocalDateTime.now(),
+                            UUID.randomUUID(),
+                            LocalDateTime.now(),
+                            UUID.randomUUID()));
+        }
+
+        when(objectMapper.readValue(
+                        ArgumentMatchers.<InputStream>any(),
+                        ArgumentMatchers.<TypeReference<List<EndorsementModelFromJSON>>>any()))
+                .thenReturn(dummyEndorsements);
+        Page<EndorsementModelFromJSON> result =
+                endorsementService.getEndorsementsFromDummyData(pageRequest, null, userID);
+
+        assertEquals(Page.empty(pageRequest), result);
```

Comment:
Yes, in the scenario you've mentioned correct. But my test has pageNumber as 10 and pageSize as 10, with 15 endorsements we will only have 2 pages. It will return an empty page if we try to access the 10th page.

Reply:
Got it 
Thanks for explaining